src/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -717,40 +713,24 @@ public virtual void Emit(OpCode opcode, long arg)
         {
             EnsureCapacity(11);
             InternalEmit(opcode);
-            m_ILStream[m_length++] = (byte)arg;
-            m_ILStream[m_length++] = (byte)(arg >> 8);
-            m_ILStream[m_length++] = (byte)(arg >> 16);
-            m_ILStream[m_length++] = (byte)(arg >> 24);
-            m_ILStream[m_length++] = (byte)(arg >> 32);
-            m_ILStream[m_length++] = (byte)(arg >> 40);
-            m_ILStream[m_length++] = (byte)(arg >> 48);
-            m_ILStream[m_length++] = (byte)(arg >> 56);
+            BinaryPrimitives.TryWriteInt64LittleEndian(m_ILStream.AsSpan(m_length), arg);
```

Comment:
Since the return from TryWriteInt64LittleEndian is not checked, it would be better to use throwing version - BinaryPrimitives.WriteInt64LittleEndian.

Reply:
(multiple places)

Reply:
In what scenario would it ever break?

Reply:
Adjusted in https://github.com/dotnet/coreclr/pull/23833/commits/069fe276f583db34a22585742c0845444b21488d